scripts/gha/test_simulator.py

```diff
@@ -595,30 +621,76 @@ def _run_android_gameloop_test(package_name, app_path, gameloop_project, retry=1
   return log
 
 
-def _install_android_app(app_path):
+def _install_android_app(app_path, retry=1):
   """Install integration_test app into the emulator."""
   args = ["adb", "install", app_path]
   logging.info("Install testapp: %s", " ".join(args))
-  subprocess.run(args=args, check=False)
+  if retry > 1:
+    try:
+      subprocess.run(args=args, check=True, timeout=_CMD_TIMEOUT)
```

Comment:
I don't really love this construction - repeating the subprocess.run on multiple lines. Is there a cleaner way to do this?

(also, like I said on iOS, WDYT about a more generic "run with retry" function?)

Reply:
I don't like this construction too. Just, for different retry, we need to run different cleaning functions. Do you have a better idea?

Reply:
What about have a set of retry functions:
run_with_retry // retry on except
run_with_retry_reboot // retry on except, reboot before retry
run_with_retry_wipe_reboot
run_test_with_retry // retry on log error info

Reply:
Hmm, are they similar enough to have a single function with a parameter that controls its behavior? (or maybe one for reboot and one for not reboot)

Reply:
All those are some tiny differences. For those in common, I created the func `_reset_emulator_on_error()`.
And, yeah. I think I could add  parameters (more than one) to control the behavior.